src/components/RegisterComponent.vue

```diff
@@ -9,167 +9,377 @@
+import { extend } from 'vee-validate';
+import { Province, Gender } from '../components/type';
+extend('numeric', {
+  ...numeric,
+  message: 'bắt buộc phải là số'
+});
+extend('requiredEmail', {
+  ...required,
+  message: 'Bắt buộc phải nhập email!'
+});
+extend('required', {
+  ...required,
+  message: 'bắt buộc phải nhập'
+});
+extend('requiredPw', {
+  ...required,
+  message: 'Bắt buộc password!'
+});
+extend('email', {
+  ...email,
+  message: 'bắt buộc phải là email'
+});
+extend('min', {
+  ...min,
+  message: 'bắt buộc 8 kí tự'
+});
+extend('alphaSpaces', {
+  message: (field) => `${field}` + ' ' + 'không được nhập dấu cách',
+  validate: (value) => {
+    for (let i = 0; i < value.length - 1; i++) {
+      let input = value.charAt(i);
+      if (input == ' ') {
+        return false;
+      }
+    }
+    return true;
+  }
+});
+extend('requiredCmnd', {
+  message: (field) => `${field}` + ' ' + 'phải 9 hoặc 12 số',
+  validate: (value) => {
+    if (value.length == 9 || value.length == 12) {
+      return true;
+    }
+    return false;
+  }
+});
+
 @Component({})
 export default class UserComponent extends Vue {
-  data() {
-    return {
-      email: ''
-    };
+  $router: any;
+  menu = false;
+  cmnd = '';
+  email = '';
+  password = '';
+  fullname = '';
+  birthday = '';
+  genders = [Gender.MALE, Gender.FEMALE];
+  gender = Gender.MALE;
+  selectWard = '';
+  selectedProvince = '';
+  selectDistrict = '';
+
+  provinces: Province[] = [
+    {
+      id: 1,
+      name: 'thaibinh',
+      districts: [
+        {
+          id: 1,
+          name: 'thaithuy',
+          wards: [
+            {
+              id: 1,
+              name: 'diem dien'
+            },
+            {
+              id: 2,
+              name: 'thuy truong'
+            }
+          ]
+        },
+        {
+          id: 2,
+          name: 'kienxuong',
+          wards: [
+            {
+              id: 1,
+              name: 'diem dien1'
+            },
+            {
+              id: 2,
+              name: 'thuy truong1'
+            }
+          ]
+        }
+      ]
+    },
+    {
+      id: 2,
+      name: 'hanoi',
+      districts: [
+        {
+          id: 1,
+          name: 'hk',
+          wards: [
+            {
+              id: 1,
+              name: 'truong dinh'
+            },
+            {
+              id: 2,
+              name: 'traica'
+            }
+          ]
+        },
+        {
+          id: 2,
+          name: 'caugiay',
+          wards: [
+            {
+              id: 1,
+              name: 'truong dinh1'
+            },
+            {
+              id: 2,
+              name: 'traica1'
+            }
+          ]
+        }
+      ]
+    }
+  ];
+  get district() {
+    return this.selectedProvince?.['districts' as unknown as any] ?? [];
+  }
+  get ward() {
+    return this.selectDistrict?.['wards' as unknown as any] ?? [];
+  }
```

Comment:
```suggestion
  get districts(): District[] {
    return this.selectedProvince?.districts ?? [];
  }
  get wards(): Ward[] {
    return this.selectDistrict?.wards ?? [];
  }
```

Chú ý convention. Array thì phải viết tên số nhiều. Type phải ghi đầy đủ, ko dc viết as như thế kia